modules/core/src/lib/layer.ts

```diff
@@ -1072,17 +1080,25 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
 
       context.device.setParametersWebGL({polygonOffset: offsets});
 
-      // Call subclass lifecycle method
-      context.device.withParametersWebGL(parameters, () => {
-        const opts = {renderPass, moduleParameters, uniforms, parameters, context};
+      const deckParameters = this.context.deck?.props.parameters;
+      const layerParameters = this.props.parameters;
+      const mergedParameters = {...deckParameters, ...layerParameters, ...parameters};
 
-        // extensions
-        for (const extension of this.props.extensions) {
-          extension.draw.call(this, opts, extension);
-        }
+      // Call subclass lifecycle method
+      const opts: DrawOptions = {
```

Comment:
Can you try calling `model.setParameters` here for all `this.getModels()`? I think the exception is rare (ScenegraphLayer comes to mind).

Reply:
Done. It works, however I cannot get picking to work if I remove the `withParametersWebGL` call below

Reply:
`model.setParameters` takes [RenderPipelineParameters](https://github.com/visgl/luma.gl/blob/c5c15fbb9ab2fcaad998d030eac9e6312dd70aad/modules/core/src/adapter/types/parameters.ts#L181) (v9 string format) and `device.withParametersWebGL` takes [GLParameters](https://github.com/visgl/luma.gl/blob/c5c15fbb9ab2fcaad998d030eac9e6312dd70aad/modules/constants/src/webgl-types.ts#L406) (v8 WebGL constant format). Picking pass used the later (`blend` and `blendColor`).